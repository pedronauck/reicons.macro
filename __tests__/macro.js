/* eslint no-undef: 0 */

const path = require('path')
const pluginTester = require('babel-plugin-tester')
const plugin = require('babel-macros')

const projectRoot = path.join(__dirname, '../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  }
})

const hasError = (message) => (err) => {
  if (err && new RegExp(message).test(err.message)) {
    return true
  }
}

pluginTester({
  plugin,
  snapshot: true,
  tests: withFilename([{
    title: 'as function',
    code: `
      const reicons = require('../reicons.macro')
      const IcTick = reicons('./tick.svg')
    `
  }, {
    title: 'with function around',
    code: `
      const reicons = require('../reicons.macro')

      const createIcon = (Icon) => {
        return <Icon />
      }

      const IcTick = createIcon(reicons('./tick.svg'))
    `
  }, {
    title: 'throw if has not .svg file extension',
    snapshot: false,
    error: hasError('You need to require a valid .svg file!'),
    code: `
      const reicons = require('../reicons.macro')
      const IcTick = reicons('./tick')
    `
  }, {
    title: 'throw if file doesn\'t exist',
    snapshot: false,
    error: hasError('Cannot find module \'./check.svg\''),
    code: `
      const reicons = require('../reicons.macro')
      const IcCheck = reicons('./check.svg')
    `
  }])
})

function withFilename(tests) {
  return tests.map(t => {
    const test = { babelOptions: { filename: __filename } }

    if (typeof t === 'string') {
      test.code = t
    }
    else {
      Object.assign(test, t)
      test.babelOptions.parserOpts = test.babelOptions.parserOpts || {}
    }

    Object.assign(test.babelOptions.parserOpts, {
      plugins: ['jsx']
    })

    return test
  })
}
