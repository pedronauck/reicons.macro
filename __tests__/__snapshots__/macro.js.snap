// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. as function 1`] = `

const reicons = require('../reicons.macro')
const IcCheck = reicons('./tick.svg')

      ↓ ↓ ↓ ↓ ↓ ↓

const IcCheck = function IcCheck(props) {
  return <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 125" className="Icon" style={{
    border: '1px solid #fff'
  }} {...props}><path d="M74.042 35.27a3.5 3.5 0 0 0-4.949.08L42 63.333 27.938 50.422a3.499 3.499 0 1 0-4.734 5.156l16.572 15.216a3.49 3.49 0 0 0 2.366.922c.916 0 1.83-.357 2.516-1.065l29.464-30.432a3.5 3.5 0 0 0-.08-4.949zM50 8.5C27.117 8.5 8.5 27.117 8.5 50S27.117 91.5 50 91.5 91.5 72.883 91.5 50 72.883 8.5 50 8.5zm0 76c-19.023 0-34.5-15.477-34.5-34.5S30.977 15.5 50 15.5 84.5 30.977 84.5 50 69.023 84.5 50 84.5z" /></svg>;
};

`;
